numpy/core/src/umath/ufunc_object.c

```diff
@@ -223,6 +223,13 @@ _find_array_method(PyObject *args, PyObject *method_name)
 static PyObject *
 _get_output_array_method(PyObject *obj, PyObject *method,
                          PyObject *input_method) {
+    if (obj == Py_Ellipsis) {
+        if (input_method != NULL) {
+            Py_INCREF(input_method);
+            return input_method;
+        }
+        Py_RETURN_NONE;
```

Comment:
I think this needs a comment saying that "returning `None` instead of `NULL` causes `PyArray_Return` not be called"